### ISSUE TITLE
[Responsiveness] Roslyn initialization is forced onto the UI thread, forcing the project system to synchronize initialize it (6.9% of solution open)

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#55297]
IWorkspaceProjectContextFactory.CreateProjectContextAsync is forced on the UI thread via Roslyn during solution load. UI thread is scarce resource during solution load and this means that Roslyn is prevented from being initialized until each project synchronously blocks on it in [PrioritizedOnAfterOpenProject](http://index/?query=WorkspaceProjectContextHost&rightProject=Microsoft.VisualStudio.LanguageServices&file=ProjectSystem%5CLanguageServices%5CWorkspaceProjectContextHostInitiator.cs&line=40). 

The project system would like to overlap this initialization so that we can make use of as many cores as possible during load.

**Comment by davkean** [comment]
I've moved it over, the project system change has been merged in.

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi @davkean I'm confused for what the remaining asks are here for this bug. It looks like @drewnoakes reactivated it when @CyrusNajmabadi's fix had DDRIT failures, but it seems all the changes have now gone in?

**Comment by davkean** [comment]
Project System removed their switch but this path immediately switches to the UI thread blocking the UI thread and preventing other things from progressing and overlapping.

**Comment by CyrusNajmabadi** [comment]
Yes.  The only work we did was to enable this to be called asynchronously from any thread.  We didn't do anything on our end to make things better on the UI thread.

**Comment by jasonmalinowski** [comment]
@drewnoakes I think we can close this at this point, or do you still have a UI thread switch before calling into Roslyn somewhere?

**Comment by davkean** [comment]
@jasonmalinowski This is representing the time Roslyn spends on the UI thread, last I saw in some traces, this was up to 30%. This isn't resolved.

**Comment by jasonmalinowski** [comment]
@davkean Can you share the trace then?

**Comment by jasonmalinowski** [comment]
@davkean Also to be clear I'm using this bug to track the original request which was making the project initialization path moved off the UI thread which is now as far as it can go without having shell APIs changed; if there's other paths that aren't that, we should have bugs for those, but let's have new bugs for those.

**Comment by davkean** [comment]
@jasonmalinowski I filed the original bug, and its always been about Roslyn chewing the UI thread. 

All Orchard Core traces are showing this. Here's from yesterday's build from a cached load:

![image](https://user-images.githubusercontent.com/1103906/159581232-10b6e2da-a618-46a4-9034-bf3392ec168c.png)

If you believe that Roslyn are unable to address these, then dupe these against the shell bugs we've filed. Looking at the trace, I see that Roslyn forces the editor package to be initialized on the UI thread, which is not helping.
